merlin/models/tf/transformers/transforms.py

```diff
@@ -37,10 +37,62 @@ class LastHiddenState(Layer):
         The output class returned by the HuggingFace transformer layer
     """
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.supports_masking = True
+
     def call(self, inputs: TFBaseModelOutputWithPoolingAndCrossAttentions):
         return inputs.last_hidden_state
 
 
+@Block.registry.register("inference_hidden_state")
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class InferenceHiddenState(Layer):
+    """A post-processing layer to select the hidden state
+    of the next-item position, during inference.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.supports_masking = True
+
+    def call(
+        self,
+        inputs: tf.Tensor,
+        training: bool = False,
+        testing: bool = False,
+        mask: tf.Tensor = None,
```

Comment:
Why do we use `supports_masking` here, and also pass `mask` in call()?
I should be one or the other I guess.
How is going to provide the mask to the `call()`?
Can't you extract the mask from `inputs._keras_mask` ?

Reply:
Os from Keras [documentation](https://www.tensorflow.org/guide/keras/masking_and_padding#supporting_masking_in_your_custom_layers), I understand that : 
* `supports_masking` is used to tell the block to forward the input mask to the following layer
* if you want to use the actual `mask` information inside your block, they explained [here](https://www.tensorflow.org/guide/keras/masking_and_padding#supporting_masking_in_your_custom_layers) that you just need to set `mask=None` in the call method. Keras automatically extracts the mask from `inputs._keras_mask` and pass it as additional param to the call function.  so using `inputs._keras_mask` is equivalent to setting `mask` as a param of the module's call. We can

Reply:
I updated the method to use the logic of  `inputs._keras_mask` instead, so as to align with other Merlin Models blocks